### PR TITLE
feat: Add base path config to app and chart

### PR DESCRIPTION
**Pull request by fhoekstra**
## Overview

Although it is best practice to use a subdomain per app, in some environments it is still desired to have apps share a (sub)domain and route traffic to each app based on base paths. This PR adds that functionality to `konflate`.

## Test evidence

I've tested both the image (with upstream chart and `config.extraEnvs`) and the chart, with and without base path, on my own repo: see https://github.com/fhoekstra/home-ops/commit/ef0bab3b5fecd4ec6c8820fa76337095fba3b47b and https://github.com/fhoekstra/home-ops/commit/3d1a8f55311877581dd95bbf2fad0f7e60d32961 for example.

## Required feedback from reviewer

- Do we want to automatically fill out the `PathPrefix` in HttpRoute matches when basePath is set or is it less error-prone to keep it explicit and leave it to the user to configure Ingess/HttpRoute for basePath?
- Do you think the Helm templates for the probes are OK? Should they be simplified?

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/home-operations/.github/blob/main/CONTRIBUTING.md)
- AI usage disclosure: YES, I have asked Kimi K3 to make a plan, K2.7 to make the changes to the code and reviewed them, then refined the generated code as well as a final review pass by K3. I have reviewed everything myself and am solely responsible for the code, regardless of the tooling I used.

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -414,12 +424,17 @@ func (s *Server) postComment(pr api.PR, st api.JobStatus) {
...
+// "" when that's unset (the status is then posted without a link). When
+// BasePath is set and PublicURL does not already end with it, BasePath is
+// appended so the two settings cannot drift silently.
 func (s *Server) reviewURL(number int) string {
 	base := strings.TrimRight(s.cfg.PublicURL, "/")
 	if base == "" {
 		return ""
 	}
+	if p := s.cfg.BasePath; p != "" && !strings.HasSuffix(base, p) {
+		base += p
```
[the remark this answers is not in the document]
Comment (fhoekstra) [reply internal/server/server.go]: I don't understand this entirely so I'm not sure this is a real issue

File: internal/config/config.go [hunk 2]
```diff
@@ -530,10 +538,39 @@ func Load() (*Config, error) {
...
+	}
+	if strings.ContainsAny(p, "?#") {
+		return fmt.Errorf("config: KONFLATE_BASE_PATH must not contain query or fragment")
+	}
+	// net/url parsing would accept "/foo/../bar" as a path; split and check each
+	// segment to reject traversal.
+	for _, seg := range strings.Split(strings.Trim(p, "/"), "/") {
+		if seg == "" || seg == "." || seg == ".." {
+			return fmt.Errorf("config: KONFLATE_BASE_PATH must not contain empty, ., or .. segments")
+		}
```
[the remark this answers is not in the document]
Comment (fhoekstra) [reply internal/config/config.go:571]: Addressed in 9514b675c6b782f496ce49f6f54a64b2f02ee222

File: charts/konflate/templates/_helpers.tpl [hunk 3]
```diff
@@ -113,3 +113,16 @@ Name of the PVC backing the source cache (existingClaim wins).
 {{- define "konflate.cacheClaimName" -}}
 {{- default (printf "%s-cache" (include "konflate.fullname" .)) (tpl (.Values.persistence.existingClaim | default "") $) -}}
 {{- end }}
+
+{{/*
+Prepend config.basePath to a probe/API path when konflate is served under a
+subpath (preserve-prefix deployment). Empty basePath returns the path unchanged.
+*/}}
+{{- define "konflate.prefixedPath" -}}
+{{- $base := .Values.config.basePath | default "" -}}
```
[the remark this answers is not in the document]
Comment (fhoekstra) [reply charts/konflate/templates/_helpers.tpl]: Addressed in 0d767fc405ff377680aba429d7a6d513b666f8c4

**Comment by onedr0p** [comment]
Thanks for the PR! I reviewed it in depth and pushed 5679fef with fixes on top of your branch (rather than a long review back-and-forth). Summary of what changed and why:

**Bugs fixed**

- `internal/web/tests/pwa.spec.ts` still asserted the old absolute manifest URLs (`/manifest.webmanifest`, `start_url: "/"`), so the Playwright suite failed deterministically. Updated to the relative forms.
- `reviewURL`'s `strings.HasSuffix(base, p)` dedupe guard ran against the full URL string, so a `publicUrl` like `https://konflate` with basePath `/konflate` matched via the scheme's second slash and silently skipped the append (posted review links 404). It now parses the URL and compares only the path component; regression test added.
- A probe `httpGet` override that omits `path` rendered `path: /platform/konflate%!s(<nil>)` (probes could never pass). It now defaults to the prefixed root.
- `config.basePath` was the only config value not passed through `tpl`, so a templated value broke the helm-test curl URL permanently. The helper now `tpl`s it like everything else.
- The bare-prefix 301 dropped the query string; it's preserved now.
- `NOTES.txt` still printed unprefixed URLs, so the post-install instructions 404'd under a base path.

**Simplifications**

- The SPA now derives its base path from `location.pathname` (the app is hash-routed, so the document URL is always `<base>/`) — the same trick your `sw.js` already used. That removes the whole `%KONFLATE_BASE_PATH%` placeholder / startup-render / `uiHandler` special-case machinery and restores the strict `script-src 'self'` CSP (no inline-script hash), and it also fixes the placeholder throwing a `SyntaxError` under `vite dev`/`vite preview`.
- The mux is built with root-relative patterns and mounted once under the prefix via `http.StripPrefix`, instead of prefixing all 16 registrations — a future route added the usual way can't forget the prefix (which would only surface in subpath deployments). `routePath` is gone.
- The three copy-pasted probe stanzas in `deployment.tpl` collapsed into one `konflate.probeSpec` helper.
- `getJSON` prefixes `basePath` once internally, so API call sites stay root-relative.

**Hardening / parity**

- The chart now normalizes `config.basePath` exactly like the binary and `fail`s at template time on values the binary would reject at boot (missing leading `/`, `..` segments, ServeMux metacharacters) — previously those rendered fine and only CrashLooped the pod. `KONFLATE_BASE_PATH` is emitted normalized.
- Dropped the manifest `id`: `"./"` resolves against the origin (unlike `start_url`/`scope`), so all instances on one host would have shared a single PWA identity. With no `id` it defaults to the resolved `start_url`, giving each deployment its own.

Verified: full Go suite, golangci-lint, all 83 helm-unittest tests, `helm template` edge cases (templated/multi-slash/invalid basePath, nil probe path, NOTES output), UI typecheck/build, and the Playwright suite. (`ui.spec.ts` "] review navigation" flakes under load on `main` too — pre-existing, unrelated.)
